### ISSUE TITLE
.luacompleterc doesn't work on Windows

### ISSUE THREAD

**Issue by nbredikhin** [dapetcu21/atom-autocomplete-lua#4]
My issue seems to be very similar to #3, but the difference is that I'm running Atom on **Windows 10 x64**. Everything works perfectly except `.luacompleterc`.

Here's my `.luacompleterc` contents:
_(I actually tried many different examples of this file, etc, but nothing works)_
```json
{
  "global": {
    "type": "table",
    "fields": {
      "myfunc":  { 
        "type": "function",
        "description": "Just testing",
        "args": [{ "name": "arg1" }, { "name": "arg2", "displayName": "[arg2]" }]
      }
    }
  }
}
```

So I tried to debug your package and here is some information I got.

1. Here you can see my test project structure and also you can see that debugger stops at breakpoint and show the correct file path.
![image](https://cloud.githubusercontent.com/assets/3995234/19911358/1a7ce268-a0a4-11e6-9a93-f9f607f8e3d2.png)

2. Here's what's being passed to `utils.mergeOptionsCached` method. We can see that it has opened and read my `.luacompleterc` with no problems: 
![image](https://cloud.githubusercontent.com/assets/3995234/19911402/6a0a3c54-a0a4-11e6-9f98-059dbdb106c7.png)

3. Then I renamed my function in `.luacompleterc` to avoid caching and checked what does mergeOptionsCached return:
![image](https://cloud.githubusercontent.com/assets/3995234/19911452/ac38ca46-a0a4-11e6-8b93-05cff5c91578.png)
![image](https://cloud.githubusercontent.com/assets/3995234/19911477/d7b6e98c-a0a4-11e6-80bc-380bed81adcb.png)

I'm not really familiar with Atom packages, but I guess that `LuaCompleteRcProvider`getOptions` returns what it's supposed to return. So do you have any ideas? Thank you in advance.

_P.S: Sorry for my bad english and thank you for this really useful package._

**Comment by dapetcu21** [comment]
Hi. Sorry for answering so late. I'll take a look on windows when I get home and see if I can reproduce.

**Comment by dapetcu21** [comment]
I was able to reproduce _something_ and fix it. I just find it weird that it manifested only on Windows. Try 0.7.1 and tell me how that works for you.

**Comment by nbredikhin** [comment]
Looks like it works now. Thank you for this fix!

Can I ask here, how can I force it to reload .luacompleterc btw? It doesn't reload when saving/reopening/editing files, but only when I restart Atom.

And another little question if possible: is/will it be possible to use named types references inside the named types definitions? Like that:
![image](https://cloud.githubusercontent.com/assets/3995234/20040189/ff0fc57c-a463-11e6-8339-8158c2cb4f7e.png)

Currently it looks like that for me:
![image](https://cloud.githubusercontent.com/assets/3995234/20040204/39bc92fe-a464-11e6-88b3-38901674e8a0.png)

_I'm not sure, maybe it should probably be moved to another issue_

**Comment by dapetcu21** [comment]
For your first question, `.luacompleterc` should get automatically reloaded. If it doesn't reload for you, then that's a bug and I'll take a look.

And as far as nested `namedTypes` go, good catch! I didn't think about that. I'll fix it in a minute.

**Comment by dapetcu21** [comment]
I opened a separate issue about the `.luacompleterc` problem. For the `namedTypes` issue, I just pushed a commit, but I'll release it along with the fix for #5.
